### PR TITLE
feat: Sign in

### PR DESCRIPTION
**Pull request by Juliocho990**
## Summary

* `POST /users/session` endpoint was created
* `email` and `password` request values were validated.
* SignIn was implemented by using `jwt-simple` as _token based authentication system_.

## Known Issues

None

## Trello Card

https://trello.com/c/0kRbPH9Y/2-sign-in

### CODE REVIEW

**Review by blueJayWays — APPROVED** [review]
Test are missing, and check why travis is failing !

**Review by blueJayWays — CHANGES_REQUESTED** [review]
Test are missing, and check why travis is failing !

File: test/users.test.js [hunk 1]
```diff
@@ -77,4 +79,52 @@ describe('Users', () => {
...
+
+    test('bad request should be returned when password does not match', async done => {
+      const getUserByEmailSpy = jest.spyOn(UsersService, 'getUserByEmail');
+      getUserByEmailSpy.mockImplementation(getUserByEmailMock);
+      signInInput.password = 'abc12345';
+      await request(app)
+        .post('/users/sessions')
+        .send(signInInput)
+        .expect('Content-Type', /json/)
+        .expect(400)
```
Comment (blueJayWays) [inline test/users.test.js]: 400 http status code means that the request was malformed. I think you should use 401 unauthorized when the login fails.

File: test/users.test.js [hunk 2]
```diff
@@ -77,4 +79,52 @@ describe('Users', () => {
...
+    });
+
+    test('bad request should be returned when user does not exist', async done => {
+      const getUserByEmailSpy = jest.spyOn(UsersService, 'getUserByEmail');
+      getUserByEmailSpy.mockImplementation(getNullUserByEmailMock);
+      await request(app)
+        .post('/users/sessions')
+        .send(signInInput)
+        .expect('Content-Type', /json/)
+        .expect(400)
```
Comment (blueJayWays) [inline test/users.test.js]: same here. 400 indicates the request was malformed.  For the requests that the BDD could't find you should use 404, BUT this is a login request, so for security reasons, you must use 401 unauthorized.

File: test/users.test.js (hunk 1 above)
Comment (Juliocho990) [reply test/users.test.js]: ok

File: test/users.test.js (hunk 2 above)
Comment (Juliocho990) [reply test/users.test.js]: ok

**Review by blueJayWays — APPROVED** [review]

**Pull request merged**